aicsimageio/readers/sldy_reader/sldy_image.py

```diff
@@ -0,0 +1,301 @@
+        interplane_spacing = self._channel_record.get("mInterplaneSpacing")
+        self.physical_pixel_size_x = micron_per_pixel / optovar_mag * x_factor
+        self.physical_pixel_size_y = micron_per_pixel / optovar_mag * y_factor
+        self.physical_pixel_size_z = (
+            float(interplane_spacing) if interplane_spacing is not None else None
+        )
+
+        data_path_matcher = fs.glob(self.image_directory / f"{data_file_prefix}*.npy")
+        self._data_paths = set([Path(data_path) for data_path in data_path_matcher])
+
+        # Create mapping of timepoint / channel to their respective data paths
+        self._timepoint_to_data_paths = SldyImage._get_dim_to_data_path_map(
+            self._data_paths, timepoint_file_prefix
+        )
+        self._channel_to_data_paths = SldyImage._get_dim_to_data_path_map(
+            self._data_paths, channel_file_prefix
+        )
+
+        # Create simple sorted list of each timepoint and channel
+        self.timepoints = sorted(self._timepoint_to_data_paths.keys())
+        self.channels = sorted(self._channel_to_data_paths.keys())
+
+    @property
+    def metadata(self) -> Dict[str, Optional[dict]]:
+        """
+        Returns a dictionary representing the metadata of this acquisition.
+
+        Returns
+        -------
+        metadata: Dict[str, dict]
+            Simple mapping of metadata file names to the metadata extracted
+            from them. Possibly different than the actual yaml due to mapping
+            the yaml to Python dictionaries, specifically with duplicate keys.
+        """
+        if self._metadata is None:
+            self._metadata = {
+                "annotation_record": SldyImage._get_yaml_contents(
+                    self._fs, self.image_directory / "AnnotationRecord.yaml", False
+                ),
+                "aux_data": SldyImage._get_yaml_contents(
+                    self._fs, self.image_directory / "AuxData.yaml", False
+                ),
+                "channel_record": self._channel_record,
+                "elapsed_times": SldyImage._get_yaml_contents(
+                    self._fs, self.image_directory / "ElapsedTimes.yaml", False
+                ),
+                "image_record": self._image_record,
+                "mask_record": SldyImage._get_yaml_contents(
+                    self._fs, self.image_directory / "MaskRecord.yaml", False
+                ),
+                "sa_position_data": SldyImage._get_yaml_contents(
+                    self._fs, self.image_directory / "SAPositionData.yaml", False
+                ),
+                "stage_position_data": SldyImage._get_yaml_contents(
+                    self._fs, self.image_directory / "StagePositionData.yaml", False
+                ),
+            }
+
+        return self._metadata
+
+    def get_data(
+        self, timepoint: Optional[int], channel: Optional[int], delayed: bool
+    ) -> np.ndarray:
```

Comment:
since this seems to be the only entry point to actually load the pixel data, it looks like it is incomplete with respect to AICSImage.get_image_data and AICSImage.get_image_dask_data.     I suspect that our reader unit tests may not be capturing that?  Somehow those function calls need to end up calling some code like this.
Typical function calls would be:
daskarr = AICSImage(sldypath, reader=SldyReader).get_image_dask_data("ZYX", T=t, C=c)
daskarr = AICSImage(sldypath, reader=SldyReader).get_image_dask_data("CZYX", T=t)
daskarr = AICSImage(sldypath, reader=SldyReader).get_image_dask_data("CYX", T=t, Z=z)
These are representative of a compute workflow that is reading the data one time sample at a time.
Then to actually resolve the dask array you would call `final_ndarry_in_memory = daskarr.compute()`

In the delayed=true case, I expected that the data should be returned as a dask array.

Reply:
I might be misinterpreting this comment, but this method is intended to grab image data as it is stored which is as `.npy` files (in ZYX or YX) split into different files by timepoint and/or channel and not meant to mirror the functionality supplied by `get_image_data` or `get_image_dask_data` on `AICSImage` per se.

Instead, what _**should**_ happen, is `AICSImage` will request the `data` or `dask_data` property loading in the full 5D dask or np array from the `SldyReader` and then will [reshape it here](https://github.com/AllenCellModeling/aicsimageio/blob/e0f9b0d47fab90526cde4a23f36ec5d7a9aff393/aicsimageio/aics_image.py#L649-L654) and when compute is called dask will figure out which nodes to resolve.

Testing this locally I ran:
```
data_with_all_z = image.get_image_dask_data("CZYX", T=0).compute()
print(data_with_all_z.size)  # yields ~15MB

data  = image.get_image_dask_data("CYX", T=0, Z=0).compute()
print(data.size)  # yields ~3MB
```

The data that included the Z dimension (`data_with_all_z`) was ~15mb, exactly 5 times bigger than the data with a specific Z dimension (`data` which was 3MB) which I think makes sense since the size of the Z dimension is 5 (3 * length of Z dimension = 15 or 3 * 5 = 15).

Reply:
I'm sorry, it's all good - I misread this and somehow completely missed SldyReader.__read_delayed calling SldyImage.get_data.  My bad!  Thanks for the detailed response.  I'm on the same page again.